src/test/kotlin/com/ampnet/identityservice/controller/Web3jControllerTest.kt

```diff
@@ -0,0 +1,36 @@
+package com.ampnet.identityservice.controller
+
+import com.ampnet.identityservice.config.ApplicationProperties
+import com.ampnet.identityservice.contract.Issuer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.web3j.crypto.Credentials
+import org.web3j.protocol.Web3j
+import org.web3j.protocol.http.HttpService
+import org.web3j.tx.gas.StaticGasProvider
+import java.math.BigInteger
+
+@ExtendWith(value = [SpringExtension::class])
+class Web3jControllerTest {
+
+    private val applicationProperties = ApplicationProperties()
+    private val web3j = Web3j.build(HttpService(applicationProperties.provider.blockchainApi))
+
+    private val privateKey = "034b74295a894357ad6e67c051009bf31de51a83df4c72751efbf7032b417ec2"
+    private val contractAddress = "0x8Ba5082E853b87E8D92970506EBb7c7097d6F640"
+    private val walletAddress = "0x745367860c5015B1E0AC04E00f1DbAd83B7dC272"
+
+    @Test
+    fun mustReturnTrueForWalletApproved() {
```

Comment:
You reading the value from an actual contract on Görli in this test right? That is fine for now, but in the future it would be better to run a local blockchain onto which you deploy a mock contract. That way you avoid having possibly flaky test due to the network issues - or worse yet, a test which starts failing because the value stored in the contract has changed on blockchain.

Reply:
Ok, this test is not there to stay, it was just as a show off that reading/writing to the smart contract works.